File: tiq-e/math/4/main.go
package main

import (
	"fmt"
	"time"
)

func romanToInt(s string) int {
	romanValue := map[rune]int{
		'I': 1,
		'V': 5,
		'X': 10,
		'L': 50,
		'C': 100,
		'D': 500,
		'M': 1000,
	}

	var arabic int

	var inputLength = len(s)
	for i, char := range s {
		if i+1 < inputLength && romanValue[char] < romanValue[rune(s[i+1])] {
			arabic -= romanValue[char]
		} else {
			arabic += romanValue[char]
		}
	}

	return arabic
}

func main() {
	n := "M"

	start := time.Now()
	result := romanToInt(n)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/math/4/main.ts
import {performance} from "perf_hooks";

function romanToInt(s: string): number {
    let map = new Map(),
        int = 0,
        i = 0;
    map.set("I", 1);
    map.set("V", 5);
    map.set("X", 10);
    map.set("L", 50);
    map.set("C", 100);
    map.set("D", 500);
    map.set("M", 1000);

    while (i < s.length) {
        const currentValue = map.get(s[i]);
        if (currentValue < map.get(s[i + 1])) int -= currentValue;
        else int += currentValue;

        i++;
    }
    return int;
}

const n = "IV";

const start = performance.now();
const result = romanToInt(n);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);
